include/aspect/parameters.h

```diff
@@ -373,6 +373,18 @@ namespace aspect
       }
     };
 
+    /**
+     * A data structure containing vectors of the indices of each
+     * compositional field type.
+     */
+    struct FieldTypeIndices
+    {
+      std::vector<unsigned int> composition;
```

Comment:
Oh, I see. Are you saying that this vector contains the indices of those fields that correspond to compositions? If so, what is the basis of this index? Within all compositional fields? Or vector components? Or vector blocks?